src/Microsoft.Azure.WebJobs.Extensions.Kafka/Output/KafkaAttribute.cs

```diff
@@ -15,6 +15,14 @@ namespace Microsoft.Azure.WebJobs.Extensions.Kafka
     [Binding]
     public sealed class KafkaAttribute : Attribute
     {
+        private int? compressionLevel;
```

Comment:
Could  you use KafkaOption to configure it?  We talked with @fbeltrao if we should go with Kafka annotation or host.json, This time we decided to go with `host.json` . We are reusing the KafkaProducer, so that we can't have different config for each producer. Also if we introduce the change for the Kafka attribute, we need to update here as well. https://github.com/Azure/azure-functions-java-library

https://github.com/Azure/azure-functions-kafka-extension/pull/218

Reply:
@TsuyoshiUshio I can do that change, but I don't think it makes much sense. The attribute is already used to configure things that are much more global in scope, like the "brokerList", and things that are on sort of the same level, such as "EnableIdempotence". There's a reason, I guess, that the KafkaProducerFactory.GetProducerConfig method gets the attribute as parameter, and I can't see the KafkaProducer being reused beyond this point. There's a comment in the code that there's a goal to use as few producers as possible, which is a good idea, but that is not implemented.

"Also if we introduce the change for the Kafka attribute, we need to update here as well. https://github.com/Azure/azure-functions-java-library"

Is there a technical reason for this, or is there just a desire to have feature parity?

Reply:
Hi @gliljas 
It is true. I have two reason for it. 

1. The compression settings will be global so that avoid customers confusion. For KafkaAttribute should have parameter that is for each producer and host.json is for global. (even if there is the past legacy is here)
2. Simply workload.  If we introduce the change, We need to update the https://github.com/Azure/azure-functions-java-library also, need to update here and test it. https://github.com/Azure/azure-functions-java-worker/tree/dev/endtoendtests/src/main/java/com/microsoft/azure/functions/endtoend . 

So that both on technical side and workload. Then I though `host.json` might be better place.  However, I'd like to listen @fbeltrao 's opinion.

Reply:
I do see you both have good points.

When creating producers we are reusing the same reference to the Librdkafka. Reusing is based on the value of the producer config which is created from the KafkaOptions (global) and Kafka attribute (function based).
So, @TsuyoshiUshio , unlike what I was suggesting you before, there is no technical requirement forcing us to make the compression global, sorry for the miss information. We can make it at function level as well, at the cost of creating additional Librdkafka producers (same applies if the batch size changes and other Kafka attribute settings).


What I don't know is:
- Will customers want to mix different compression settings in same function deployment?
- Is it easier for customers to set once at the global level or set always on function level?

If we don't have a clear answer perhaps being more flexible is better (defining at the function level). We should also validate and document what we recommend in terms of base producers / function.

Reply:
I think that any settings that aren't explicitly global in their nature should be possible to set on a function/attribute level, but having a fallback to a global setting is very good. In fact, I would say that broker connection things (authentication etc.) are good candidates for global settings, whereas "topic" is not and things like compression could work in both places.

Reply:
Thank you @fbeltrao That's make sense. For the feature expansion, it can be flexible. Let's go with Attribute. 
As @gliljas suggested, we might think to make it clear which parameter is global and which is not. In the future release, when major change introduced, we can make the change. Until then I'll find time to add comments on the Attribute.
Sure Let's go with Attribute!